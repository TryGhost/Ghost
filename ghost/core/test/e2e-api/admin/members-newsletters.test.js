const {agentProvider, mockManager, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyContentVersion, anyEtag, anyObjectId, anyUuid, anyISODateTime, anyArray} = matchers;
const settingsHelpers = require('../../../core/server/services/settings-helpers');
const sinon = require('sinon');

const memberMatcherShallowIncludesForNewsletters = {
    id: anyObjectId,
    uuid: anyUuid,
    created_at: anyISODateTime,
    updated_at: anyISODateTime,
    subscriptions: anyArray,
    labels: anyArray,
    newsletters: anyArray
};

let agent;

describe('Members API - With Newsletters', function () {
    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('newsletters', 'members:newsletters');
        await agent.loginAsOwner();
    });

    beforeEach(function () {
        sinon.stub(settingsHelpers, 'createUnsubscribeUrl').returns('http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme');
    });

    afterEach(function () {
        mockManager.restore();
    });

    // List Members

    it('Can fetch members who are subscribed', async function () {
        await agent
            .get('/members/?filter=newsletters.status:active')
            .expectStatus(200)
            .matchBodySnapshot({
                members: new Array(6).fill(memberMatcherShallowIncludesForNewsletters)
            })
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                etag: anyEtag
            });
    });

    it('Can fetch members who are NOT subscribed', async function () {
        await agent
            .get('/members/?filter=newsletters.status:-active')
            .expectStatus(200)
            .matchBodySnapshot({
                members: new Array(2).fill(memberMatcherShallowIncludesForNewsletters)
            })
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                etag: anyEtag
            });
    });
});

describe('Members API - With Newsletters - compat mode', function () {
    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('newsletters', 'members:newsletters');
        await agent.loginAsOwner();
    });

    beforeEach(function () {
        sinon.stub(settingsHelpers, 'createUnsubscribeUrl').returns('http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme');
    });

    afterEach(function () {
        mockManager.restore();
    });

    // List Members

    it('Can fetch members who are subscribed', async function () {
        await agent
            .get('/members/?filter=subscribed:true')
            .expectStatus(200)
            .matchBodySnapshot({
                members: new Array(6).fill(memberMatcherShallowIncludesForNewsletters)
            })
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                etag: anyEtag
            });
    });

    it('Can fetch members who are NOT subscribed', async function () {
        await agent
            .get('/members/?filter=subscribed:false')
            .expectStatus(200)
            .matchBodySnapshot({
                members: new Array(2).fill(memberMatcherShallowIncludesForNewsletters)
            })
            .matchHeaderSnapshot({
                'content-version': anyContentVersion,
                etag: anyEtag
            });
    });
});
