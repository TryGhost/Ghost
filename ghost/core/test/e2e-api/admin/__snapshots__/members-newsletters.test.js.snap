// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members API - With Newsletters - compat mode Can fetch members who are NOT subscribed 1: [body] 1`] = `
Object {
  "members": Array [
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "with-product@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Dana Barrett",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": false,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "vip-paid@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Peter Venkman",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": false,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
  ],
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 2,
    },
  },
}
`;

exports[`Members API - With Newsletters - compat mode Can fetch members who are NOT subscribed 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "2407",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Members API - With Newsletters - compat mode Can fetch members who are subscribed 1: [body] 1`] = `
Object {
  "members": Array [
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "vip@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Winston Zeddemore",
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "comped@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Vinz Clortho",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "trialing@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Ray Stantz",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "paid@test.com",
      "email_count": 0,
      "email_open_rate": 80,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Egon Spengler",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "member2@test.com",
      "email_count": 0,
      "email_open_rate": 50,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": null,
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "member1@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Mr Egg",
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
  ],
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 6,
    },
  },
}
`;

exports[`Members API - With Newsletters - compat mode Can fetch members who are subscribed 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "8506",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Members API - With Newsletters Can fetch members who are NOT subscribed 1: [body] 1`] = `
Object {
  "members": Array [
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "with-product@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Dana Barrett",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": false,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "vip-paid@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Peter Venkman",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": false,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
  ],
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 2,
    },
  },
}
`;

exports[`Members API - With Newsletters Can fetch members who are NOT subscribed 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "2407",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Members API - With Newsletters Can fetch members who are subscribed 1: [body] 1`] = `
Object {
  "members": Array [
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "vip@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Winston Zeddemore",
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "comped@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Vinz Clortho",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "trialing@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Ray Stantz",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "paid@test.com",
      "email_count": 0,
      "email_open_rate": 80,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Egon Spengler",
      "newsletters": Any<Array>,
      "note": null,
      "status": "paid",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "member2@test.com",
      "email_count": 0,
      "email_open_rate": 50,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": null,
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
    Object {
      "avatar_image": null,
      "comped": false,
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "email": "member1@test.com",
      "email_count": 0,
      "email_open_rate": null,
      "email_opened_count": 0,
      "email_suppression": Object {
        "info": null,
        "suppressed": false,
      },
      "geolocation": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "labels": Any<Array>,
      "last_seen_at": null,
      "name": "Mr Egg",
      "newsletters": Any<Array>,
      "note": null,
      "status": "free",
      "subscribed": true,
      "subscriptions": Any<Array>,
      "unsubscribe_url": "http://domain.com/unsubscribe/?uuid=memberuuid&key=abc123dontstealme",
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
    },
  ],
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 6,
    },
  },
}
`;

exports[`Members API - With Newsletters Can fetch members who are subscribed 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "8506",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
